boa/src/value/mod.rs

```diff
@@ -897,6 +911,90 @@ impl JsValue {
             Ok(false)
         }
     }
+
+    /// Retrieves value of specific property, when the value of the property is expected to be a function.
+    ///
+    /// More information:
+    /// - [EcmaScript reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-getmethod
+    pub(crate) fn get_method<K>(&self, context: &mut Context, key: K) -> JsResult<JsValue>
+    where
+        K: Into<PropertyKey>,
+    {
+        // 1. Assert: IsPropertyKey(P) is true.
+        // 2. Let func be ? GetV(V, P).
+        let func = self.get_v(context, key)?;
+
+        // 3. If func is either undefined or null, return undefined.
+        if func.is_null_or_undefined() {
+            return Ok(JsValue::undefined());
+        }
+
+        // 4. If IsCallable(func) is false, throw a TypeError exception.
+        if !func.is_callable() {
+            Err(context
+                .construct_type_error("value returned for property of object is not a function"))
+        } else {
+            // 5. Return func.
+            Ok(func)
+        }
+    }
+
+    /// The `GetV ( V, P )` abstract operation
+    ///
+    /// Retrieves the value of a specific property of an ECMAScript language value. If the value is
+    /// not an object, the property lookup is performed using a wrapper object appropriate for the
+    /// type of the value.
+    ///
+    /// More information:
+    /// - [EcmaScript reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-getmethod
+    #[inline]
+    pub(crate) fn get_v<K>(&self, context: &mut Context, key: K) -> JsResult<JsValue>
+    where
+        K: Into<PropertyKey>,
+    {
+        // 1. Let O be ? ToObject(V).
+        let o = self.to_object(context)?;
+
+        // 2. Return ? O.[[Get]](P, V).
+        o.get(key, context)
+    }
+
+    /// It determines if the value is a callable function with a `[[Call]]` internal method.
+    ///
+    /// More information:
+    /// - [EcmaScript reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-iscallable
+    #[track_caller]
+    pub(crate) fn is_callable(&self) -> bool {
+        if let Self::Object(obj) = self {
+            obj.is_callable()
+        } else {
+            false
+        }
+    }
+
+    /// Determines if `value` inherits from the instance object inheritance path.
+    ///
+    /// More information:
+    /// - [EcmaScript reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-ordinaryhasinstance
+    pub(crate) fn ordinary_has_instance(
+        &self,
+        context: &mut Context,
+        value: &JsValue,
+    ) -> JsResult<bool> {
+        if let Self::Object(obj) = self {
+            obj.ordinary_has_instance(context, value)
+        } else {
+            Ok(false)
+        }
+    }
```

Comment:
There's a `JsValue` impl at the bottom of `operations.rs` with all the methods of `JsValue` that operate on `JsObjects`. Maybe this should be put there just to have all the object related operations in the same file.

Reply:
On this, I think we should add the `is_array()` and so on methods to `JsValue`, and also the `as_array()` and so on. This would make it easier to only use `JsValue` in most places.

Reply:
@jedel1043  I think this is solved right?

@Razican Should I open an issue for implementing all the `is` and `as` methods on `JsValue` and looking for some optimizations around the codebase?

Reply:
> I think this is solved right?

Yep, I moved this to `operations.rs`